zingo-sync/src/scan.rs

```diff
@@ -201,11 +214,32 @@ where
     .await
     .unwrap();
 
+    let WitnessData {
+        sapling_initial_position,
+        orchard_initial_position,
+        sapling_leaves_and_retentions,
+        orchard_leaves_and_retentions,
+    } = witness_data;
+
+    let sapling_located_trees = tokio::task::spawn_blocking(move || {
```

Comment:
Is there any reason not to join these two futures instead of awaiting them individually, so that both tasks can spawn and run concurrently on different threads?

Reply:
as each of these tasks create concurrent synchronous tasks across multiple threads I didn't think this would improve performance but now you say it there may be some downtime where sapling_trees is almost complete and the threads can be utilised for orchard_trees so will make this change in a future PR. good find!